### PR TITLE
Include crop year & crop calendar in Purchase Order

### PR DESCRIPTION
**Pull request by riadmen**
https://github.com/metasfresh/metasfresh/issues/16510

### CODE REVIEW

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_Order.java [hunk 1]
```diff
@@ -128,4 +128,28 @@ private void invokeHandlerForEachLine(
...
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE },
+			ifColumnsChanged = { I_C_Order.COLUMNNAME_C_Harvesting_Calendar_ID, I_C_Order.COLUMNNAME_Harvesting_Year_ID })
+	public void validateHarvestingDatesForPurchaseOrder(@NonNull final I_C_Order orderRecord)
+	{
+		final SOTrx soTrx = SOTrx.ofBoolean(orderRecord.isSOTrx());
+		if (soTrx.isSales())
+		{
+			return;
+		}
+
```
Comment (cristinamghita) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_Order.java:142]: Rewrite directly:
if (orderRecord.isSOTrx())
{
return;
}

**Review by cristinamghita — CHANGES_REQUESTED** [review]
check my comment

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_Order.java [hunk 2]
```diff
@@ -101,8 +101,7 @@ public void afterReactivate(@NonNull final I_C_Order orderRecord)
 			ifColumnsChanged = { I_C_Order.COLUMNNAME_C_Harvesting_Calendar_ID, I_C_Order.COLUMNNAME_Harvesting_Year_ID })
 	public void validateHarvestingDatesForSalesOrder(@NonNull final I_C_Order orderRecord)
 	{
-		final SOTrx soTrx = SOTrx.ofBoolean(orderRecord.isSOTrx());
-		if (soTrx.isPurchase())
+		if (orderRecord.isSOTrx())
 		{
```
Comment (cristinamghita) [inline backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_Order.java:105]: I think you shall negate here: ```if (!orderRecord.isSOTrx())```

**Review by cristinamghita — CHANGES_REQUESTED** [review]

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_Order.java [hunk 3]
```diff
@@ -128,4 +127,28 @@ private void invokeHandlerForEachLine(
...
 	}
+
+	@CalloutMethod(columnNames = { I_C_Order.COLUMNNAME_C_Harvesting_Calendar_ID, I_C_Order.COLUMNNAME_Harvesting_Year_ID })
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW, ModelValidator.TYPE_BEFORE_CHANGE },
+			ifColumnsChanged = { I_C_Order.COLUMNNAME_C_Harvesting_Calendar_ID, I_C_Order.COLUMNNAME_Harvesting_Year_ID })
+	public void validateHarvestingDatesForPurchaseOrder(@NonNull final I_C_Order orderRecord)
+	{
+		final SOTrx soTrx = SOTrx.ofBoolean(orderRecord.isSOTrx());
+		if (soTrx.isSales())
+		{
```
[the remark this answers is not in the document]
Comment (riadmen) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_Order.java:137]: done

File: backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_Order.java (hunk 2 above)
Comment (riadmen) [reply backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interceptor/C_Order.java:105]: done

**Review by cristinamghita — APPROVED** [review]
lgtm

**Pull request merged**